ddtrace/opentracer/tracer.py

```diff
@@ -71,35 +75,123 @@ def __init__(self, service_name=None, config=None, scope_manager=None):
+            ownership of that dictionary, because the Tracer may use it as-is
+            to avoid extra data copying.
+        :param start_time: an explicit Span start time as a unix timestamp per
+            time.time()
+        :param ignore_active_span: an explicit flag that ignores the current
+            active `Scope` and creates a root `Span`.
+        :return: an already-started Span instance.
+        """
         pass
 
     def start_span(self, operation_name=None, child_of=None, references=None,
                    tags=None, start_time=None, ignore_active_span=False):
-        """"""
-        pass
+        """Starts and returns a new Span representing a unit of work.
+
+        Starting a root Span (a Span with no causal references)::
+            tracer.start_span('...')
+
+        Starting a child Span (see also start_child_span())::
+            tracer.start_span(
+                '...',
+                child_of=parent_span)
+        Starting a child Span in a more verbose way::
+            tracer.start_span(
+                '...',
+                references=[opentracing.child_of(parent_span)])
+        :param operation_name: name of the operation represented by the new
+            span from the perspective of the current service.
+        :param child_of: (optional) a Span or SpanContext instance representing
+            the parent in a REFERENCE_CHILD_OF Reference. If specified, the
+            `references` parameter must be omitted.
+        :param references: (optional) a list of Reference objects that identify
+            one or more parent SpanContexts. (See the Reference documentation
+            for detail)
+        :param tags: an optional dictionary of Span Tags. The caller gives up
+            ownership of that dictionary, because the Tracer may use it as-is
+            to avoid extra data copying.
+        :param start_time: an explicit Span start time as a unix timestamp per
+            time.time()
+        :param ignore_active_span: an explicit flag that ignores the current
+            active `Scope` and creates a root `Span`.
+        :return: an already-started Span instance.
+        """
+        ot_parent = child_of      # 'ot_parent' is more readable than 'child_of'
+        ot_parent_context = None  # the parent span's context
+        dd_parent = None          # the child_of to pass to the ddtracer
+
+        # Okay so here's the deal for ddtracer.start_span:
+        #  - whenever child_of is not None ddspans with parent-child relationships
+        #    will share a ddcontext which maintains a hierarchy of ddspans for
+        #    the execution flow
+        #  - when child_of is a ddspan then the ddtracer uses this ddspan to create
+        #    the child ddspan
+        #  - when child_of is a ddcontext then the ddtracer uses the ddcontext to
+        #    get_current_span() for the parent
+        if ot_parent is None and not ignore_active_span:
```

Comment:
if we pass a `child_of`, this must override the scope manager in any case. This block must be the fallback in case a `child_of` is not present.

Reply:
Alright. Let's give `child_of` precedence. (We should make note for the OpenTracing meeting to mention this in the specification so that it is defined)

Reply:
Oh wait. Currently in this block if `ot_parent == child_of` is defined then we will use that instead of the scope manager. Only if `ot_parent is None` do we ever use the scope manager.